wordpress-mobile/WordPress-Android
# WP Stories integration phase 1 - lint fixes not coming from Stories source code

This PR builds on top of #11864 

Note: the instructions for building are in the base PR #11864.

This PR fixes a lot of lint errors that appeared only after adding the `portkey-android` repo as a `git submodule` in #11864, and that are not strictly related to the submodule being added, but are all part of the code available in the base `develop` branch. I have been trying to understand why these issues are reported by the linter only after adding the submodule (and these errors do _not_ appear in the submodule's code), so after struggling for a couple of days and verifying all the libraries used are the same versions, etc as in the target codebase, decided to go ahead and fix the issues with this PR.
In this sense, while this PR needs to be in the chain of a feature branch for development, is not strictly related to the functionality that the feature branch is going to end up having, but only fixes problems in the target codebase to give the new submodule a proper welcome.

The issues that appear here were of 2 kinds:
- `UseRequireInsteadOfGet`, in order to actually fix this we'd need to change most of the `context!!` and `activity!!` declarations, which I consider this is a nice to have to make code nicer but not really a behavior change, hence I added the flag as a `warning` class in the `lint.xml` configuration in 531d608.
- use `lifeCycleOwner` in all observe() calls where a Fragment or Activity was being added for `ViewModel` changes listening. I believe this change needed to be done as it was the source of many other issues that have been otherwise fixed as they manifested themselves (see for example this one here https://github.com/wordpress-mobile/WordPress-Android/pull/10954 by @develric, I've seen other comments by @planarvoid on other PRs but couldn't find their reference again). This is fixed in a889ee4.

For the second one, I found this  SO answer here to be useful https://stackoverflow.com/questions/59521691/use-viewlifecycleowner-as-the-lifecycleowner

From the documentation of Fragment's `getLifecycleOwner` source code itself it seems this should be the way to go anyway, and is safe to change to the recommended way:
>      * Get a {@link LifecycleOwner} that represents the {@link #getView() Fragment's View}
>     * lifecycle. In most cases, this mirrors the lifecycle of the Fragment itself, but in cases
>     * of {@link FragmentTransaction#detach(Fragment) detached} Fragments, the lifecycle of the


To test:
- build and smoke test the app. Please follow the instructions in  #11864  for initialization of the submodule
- verify the changes in this PR address all of the lint errors that can be seen in the base branch PR (#11864)

PR submission checklist:
